Box_intersection_d/include/CGAL/Box_intersection_d/Box_d.h

```diff
@@ -36,14 +38,16 @@ namespace Box_intersection_d {
 template <class Dummy>
 struct Unique_numbers {
     typedef std::size_t ID;
-    Unique_numbers() : i(n++) {}
+    Unique_numbers() {
+      CGAL_STATIC_THREAD_LOCAL_VARIABLE(std::size_t, n, 0);
+      i = n++;
+    }
```

Comment:
Similarly here, I think it should be a `tbb::atomic<std::size_t>`

cc @afabri @lrineau